float32 wall_distance
float32 wall_angle
